src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/DirectoryDAO.scala

```diff
@@ -19,18 +19,18 @@ trait DirectoryDAO {
 
   def loadSubjectFromEmail(email: String): Future[Option[WorkbenchSubject]]
 
-  def createUser(user: WorkbenchUser): Future[WorkbenchUser]
-  def loadUser(userId: WorkbenchUserId): Future[Option[WorkbenchUser]]
-  def loadUsers(userIds: Set[WorkbenchUserId]): Future[Seq[WorkbenchUser]]
-  def deleteUser(userId: WorkbenchUserId): Future[Unit]
+  def createUser(user: WorkbenchPerson): Future[WorkbenchPerson]
+  def loadUser(userId: WorkbenchSubject): Future[Option[WorkbenchPerson]]
+  def loadUsers(userIds: Set[WorkbenchSubject]): Future[Seq[WorkbenchPerson]]
+  def deleteUser(userId: WorkbenchSubject): Future[Unit]
 
-  def listUsersGroups(userId: WorkbenchUserId): Future[Set[WorkbenchGroupName]]
-  def listFlattenedGroupUsers(groupName: WorkbenchGroupName): Future[Set[WorkbenchUserId]]
+  def listUsersGroups(userId: WorkbenchSubject): Future[Set[WorkbenchGroupName]]
+  def listFlattenedGroupUsers(groupName: WorkbenchGroupName): Future[Set[WorkbenchSubject]]
   def listAncestorGroups(groupName: WorkbenchGroupName): Future[Set[WorkbenchGroupName]]
 
-  def enableUser(userId: WorkbenchUserId): Future[Unit]
-  def disableUser(userId: WorkbenchUserId): Future[Unit]
-  def isEnabled(userId: WorkbenchUserId): Future[Boolean]
+  def enableUser(userId: WorkbenchSubject): Future[Unit]
+  def disableUser(userId: WorkbenchSubject): Future[Unit]
+  def isEnabled(userId: WorkbenchSubject): Future[Boolean]
```

Comment:
I am thinking the enable functions here actually belong at the service level. Enable and disable should add or remove both the user and pet. These functions should be renamed. Better yet we should just use the other group functions (but can't yet because the enabled users group has an object class of groupOfNames not groupOfUniqueNames).

Reply:
Cool, good idea. So 
```
def enableUser(userId: WorkbenchUserId): Future[Unit]
```
would enable both the user and its pet (if one exists).

Reply:
yes, the same for disable... a user should not be disabled yet have an enabled pet, right?

Reply:
A rogue pet... yeah I can't think of a use case for that. Might as well make enable/disable work atomically for users and pets.